WordPress/UITestsFoundation/Screens/ReaderScreen.swift

```diff
@@ -194,23 +210,54 @@ public class ReaderScreen: ScreenObject {
     }
 
     @discardableResult
-    public func verifySavedPosts(state: String, postLabel: String? = nil) -> Self {
+    public func verifySavedPosts(state: String, postLabel: String? = nil, file: StaticString = #file, line: UInt = #line) -> Self {
         if readerTable.cells.count > 0 {
-            XCTAssertEqual(readerTable.cells.firstMatch.label, postLabel, "Post displayed does not match saved post!")
-            XCTAssertEqual(readerTable.cells.count, 1, "There should only be 1 post!")
-            XCTAssertEqual(state, .withSavedPosts)
+            XCTAssertTrue(readerTable.cells.firstMatch.waitForExistence(timeout: 3), file: file, line: line)
+            XCTAssertEqual(readerTable.cells.firstMatch.label, postLabel, .postNotEqualSavedPostError, file: file, line: line)
+            XCTAssertEqual(readerTable.cells.count, 1, .postNotEqualOneError, file: file, line: line)
+            XCTAssertEqual(state, .withSavedPosts, file: file, line: line)
         } else {
-            XCTAssertTrue(noResultsView.waitForExistence(timeout: 3))
-            XCTAssertTrue(readerTable.label == .emptyListLabel)
-            XCTAssertEqual(state, .withoutSavedPosts)
+            XCTAssertTrue(noResultsView.waitForExistence(timeout: 3), file: file, line: line)
+            XCTAssertTrue(readerTable.label == .emptyListLabel, file: file, line: line)
+            XCTAssertEqual(state, .withoutSavedPosts, file: file, line: line)
         }
 
         return self
     }
+
+    public func likeFirstPost() -> Self {
+        likeButton.firstMatch.waitForIsHittable()
+        XCTAssertTrue(likeButton.firstMatch.label.hasPrefix(.postNotLiked))
+        likeButton.firstMatch.tap()
+
+        return self
+    }
+
+    public func verifyPostLikedOnFollowing(file: StaticString = #file, line: UInt = #line) -> Self {
```

Comment:
For consistency with `verifyPostLikedOnLikesTab`:

```suggestion
    public func verifyPostLikedOnFollowingTab(file: StaticString = #file, line: UInt = #line) -> Self {
```